### PR TITLE
A few more fixes and updates to TriBITS for transition to modern CMake (#299)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
See the individual commit messages for details.

## Notes to Reviewers

* Please review these commits one at a time and look at the commit messages as well.

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
CC: @keitat 

Hello @KyleFromKitware, can you please do a quick review of this?  This needs to be merged to TriBITS so that I can update the snapshot of TriBITS into PR trilinos/Trilinos#10614.

**Comment by bartlettroscoe** [comment]
@rppawlo, if you are reviewing these changelog entries anyway, you mind taking a look at the new CHANGELOG.md entry I added in this PR?  This is impacting Trilinos and might impact Drekar and other projects using TriBITS if they define their own TPLs (and the TPLs have dependencies on each other).

**Comment by bartlettroscoe** [comment]
Hello @KyleFromKitware, I added the commit https://github.com/TriBITSPub/TriBITS/pull/492/commits/2a6c1ceb222283de04c181a0a1db3d5a126d0fbc to assert the `.framework` extension for files on APPLE=TRUE that don't start with 'lib'.

Please review this PR.  Thanks!

**Comment by bartlettroscoe** [comment]
@KyleFromKitware, do you have any other review comments other than what is in https://github.com/bartlettroscoe/TriBITS/commit/2a6c1ceb222283de04c181a0a1db3d5a126d0fbc#commitcomment-77219732?  (Note that that comment does not show up in this PR for some reason.  You can only see it by going directly to the commit on GitHub with the that link given in the prior sentence.)

File: tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake [hunk 1]
```diff
@@ -567,13 +567,19 @@ function(tribits_external_package_get_libname_from_full_lib_path  full_lib_path
     if (beginsWithLib)
       string(SUBSTRING "${full_libname}" 3 -1 libname)
     else()
-      set(libname "${full_libname}")
+      # Must be a framework dir with extension .framework
+      get_filename_component(last_ext "${full_lib_path}" LAST_EXT)
+      if (last_ext  STREQUAL ".framework")
```
Comment (bartlettroscoe) [inline tribits/core/package_arch/TribitsExternalPackageWriteConfigFile.cmake:572]: Hello @KyleFromKitware, here is where a comment in a commit is associated with the PR and therefore will be displayed with the PR.  This is where the commit https://github.com/bartlettroscoe/TriBITS/commit/2a6c1ceb222283de04c181a0a1db3d5a126d0fbc#commitcomment-77219732 would go.

**Review by rabartlett1972 — APPROVED** [review]
Just reapproving after rebase (prior approved [above](https://github.com/TriBITSPub/TriBITS/pull/492#issuecomment-1170232067)).

**Comment by rppawlo** [comment]
@bartlettroscoe - thanks for the heads up on these changes. I'll mention them to the empire, charon and drekar teams this week. We did have a successful merge of empire last night so it seems to be working fine!

**Pull request merged**